### PR TITLE
refactor(auth): restructure authentication logic and remove unused op…

### PR DESCRIPTION
**Pull request by wraith2009**
…tions

### CODE REVIEW

File: app/(auth)/sign-in/page.jsx [hunk 1]
```diff
@@ -53,6 +53,8 @@ export default function SignInForm() {
         password: data.password,
       });
 
+      console.log(result);
```
Comment (dinxsh) [inline app/(auth)/sign-in/page.jsx:56]: forgot to omit this

File: lib/authOptions.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
+import { NextAuthOptions } from "next-auth";
+import bcrypt from "bcryptjs";
+import CredentialsProvider from "next-auth/providers/credentials";
+import GoogleProvider from "next-auth/providers/google";
+import dbConnect from "./dbConnect";
+import UserModel from "../model/User";
+
+const bypassCredentials = {
+  email: "test@gmail.com",
```
Comment (dinxsh) [inline lib/authOptions.js:9]: need to remove this, cant belong in production

File: lib/authOptions.js [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+export const authOptions = {
+  providers: [
+    CredentialsProvider({
+      name: "Credentials",
+      credentials: {
+        email: { label: "Email", type: "text", placeholder: "demo@demo.com" },
+        password: { label: "Password", type: "password" },
+      },
+      async authorize(credentials) {
+        console.log("Received credentials:", credentials);
```
Comment (dinxsh) [inline lib/authOptions.js:22]: omit print

File: model/User.js [hunk 4]
```diff
@@ -25,10 +25,6 @@ const userSchema = new Schema({
     type: String,
     default: "",
   },
-  discordId: {
```
Comment (dinxsh) [inline model/User.js:28]: need discord id for future integration keep it in the model by default can be none

Comment (wraith2009) [reply model/User.js:28]: readded this

**Pull request merged**